README.md

```diff
@@ -44,46 +50,72 @@ class ExampleConfigSection(ConfigSectionBase):
 class ExampleConfig(ConfigBase):
     """Config for an example"""
 
-    section1: ExampleConfigSection = ExampleConfigSection()
-
-    @staticmethod
-    def get_app_basename() -> str:
-        """Return the application base name.
-
-        This name, with a preceding dot, will also be the folder name in the home directory
-        that will store the config. I.e., for the case here: ~/.example
-        """
-        return "example"
+    section1: ExampleConfigSection1 = ExampleConfigSection1()
 
 ```
 
-Note that you have to define `get_app_basename()` for the container.
-
 ### Write a config file
 
-For the example, the config file `~/.example/config.toml` will be something like this:
+For the example, the config file `~/.example/config.toml` could be something like this:
 
 ```toml
 [section1]
 field1 = "my own version of field1"
 field2 = 22
 ```
 
-Currently, there is no check on types when loading the config file (to be added soon).
-Consider the case where you would use the following config file for the example:
-
-```toml
-[section1]
-field1 = "my own version of field1"
-field2 = "22"
-```
-
-Now `the_config.section1.field2` silently becomes a *string* rather than the specified int.
+The order of sections and/or fields in the toml file does not have to adhere to the order
+in which attributes have been specified in the Config(Section) classes.
 
 ### Use config parameters in your code
 
 ```python
+# the first invocation of get() will create the singleton instance of ExampleConfig
 the_config = ExampleConfig.get()
 a_variable: str = the_config.section1.field1  # a_variable == "my own version of field1"
 another_variable: int = the_config.section1.field2  # another_variable == 22
+
+# you can reload a config and / or set a non-default path
+another_config = ExampleConfig.get(reload=True, configfile_path="./my_config.tml")
+
 ```
+
+## Handling deviations in the config file
+
+### When your config file does not adhere to the specified types
+
+When loading the config file, the values specified are coerced into the appropriate type
+where possible. If type coercion is not possible, then a `pydantic.ValidationError`
+is raised. Consider the case where you would use the following config file for 
+the `ExampleConfig` defined above:
+
+```toml
+[section1]
+field1 = True
```

Comment:
I think toml doesn't support `True`, but you should use `true`. The syntax highlighting at least is very weird here.

Reply:
Right.